cygnus-arcgis/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIArcgisFeatureTableSink.java

```diff
@@ -0,0 +1,789 @@
+            throw new CygnusCappingError(e.getMessage());
+        }
+    } // capRecords
+
+    @Override
+    public void expirateRecords(long expirationTime) throws CygnusExpiratingError {
+        LOGGER.debug("[" + this.getName() + "] Expirating records (time=" + expirationTime + ")");
+
+    } // expirateRecords
+
+    /**
+     * Persist Aggregation.
+     * 
+     * @param aggregator
+     * @throws CygnusRuntimeError
+     */
+    public void persistAggregation(NGSIArcgisAggregator aggregator) throws CygnusRuntimeError {
+        try {
+            List<ArcgisAggregatorDomain> aggregationList = aggregator.getListArcgisAggregatorDomain();
+            LOGGER.debug("[" + this.getName() + "] persisting aggregation, "
+                    + aggregator.getListArcgisAggregatorDomain().size() + " features.");
+            for (ArcgisAggregatorDomain aggregation : aggregationList) {
+                String featureTableUrl = aggregation.getFeatureTableUrl();
+
+                boolean isNewFeatureTable = !arcgisPersistenceBackend.containsKey(featureTableUrl);
+                LOGGER.debug("[" + this.getName() + "] persistAggregation - Feature table: " + featureTableUrl
+                        + " is new: " + isNewFeatureTable);
+                ArcgisFeatureTable featureTable = getPersistenceBackend(featureTableUrl);
+
+                // If it's a new one, sets uniqueField value
+                if (isNewFeatureTable) {
+                    LOGGER.debug("[" + this.getName() + "] Created new backend for " + featureTableUrl
+                            + " with uniqueField: " + aggregation.getUniqueField());
+                    featureTable.setUniqueField(aggregation.getUniqueField());
+                }
+
+                featureTable.addToBatch(aggregation.getFeature());
+            }
+        } catch (CygnusRuntimeError e) {
+            String stackTrace = ExceptionUtils.getFullStackTrace(e);
+            LOGGER.debug(" PersistAggregation Error: " + stackTrace);
+            throw (e);
+        } catch (Exception e) {
+            LOGGER.error("[" + this.getName() + "] Error persisting batch, " + e.getClass().getSimpleName() + " - "
+                    + e.getMessage());
+            throw new CygnusRuntimeError(e.getMessage());
+        }
+    } // persistAggregation
+
+    /**
+     * @author dmartinez
+     *
+     */
+    public class NGSIArcgisAggregator {
+
+        private List<ArcgisAggregatorDomain> listArcgisAggregatorDomain = new ArrayList<ArcgisAggregatorDomain>();
+        private boolean enableNameMappings = true;
+        private String argisServiceUrl = "";
+
+        private NGSIArcgisAggregator() {
+        } // ArcgisAggregator
+
+        /**
+         * Constructor.
+         * 
+         * @param argisServiceUrl
+         * @param enableNameMappings
+         */
+        public NGSIArcgisAggregator(String argisServiceUrl, boolean enableNameMappings) {
+            this();
```

Comment:
Is this necessary? The constructor that matches with that signature seems to be empty.

Reply:
It's no empty anymore.

https://github.com/Stdri/fiware-cygnus/blob/c8d4419a2cf00210fcaf968d5ed96375ce2828cd/cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIArcgisFeatureTableSink.java#L407-L411